serialize.go

```diff
@@ -135,13 +135,14 @@ const maxRetryDuration = 1 * time.Second
 func lockFile(path string) (io.Closer, error) {
 	retry := 100 * time.Microsecond
 	startTime := time.Now()
+	var err error
 	for {
 		if locker, err := filelock.Lock(path); err == nil {
 			return locker, nil
 		}
 		total := time.Since(startTime)
 		if total > maxRetryDuration {
-			return nil, errgo.New("file locked for too long; giving up")
+			return nil, errgo.Notef(err, "file locked for too long; giving up")
```

Comment:
This isn't going to work. err is never set.

You want something like:

    for {
         locker, err := filelock.Lock(path)
         if err == nil {
              return locker, nil
         }

etc